bundles/com.zeligsoft.domain.dds4ccm/src/com/zeligsoft/domain/dds4ccm/utils/DDS4CCMMigrationModelTypeUtil.java

```diff
@@ -69,35 +92,268 @@ private static boolean migrateAll(Model model, boolean dryrun) {
+		
+		String pathmapConnectorLib = getPathmapConnectorLib(defaultConnectorType);
+		URI uriConnectorLib = URI.createURI(pathmapConnectorLib);
+		Package packageConnectorLib = getImportedConnectorPackage(model, uriConnectorLib);
+		ConnectorDef connectorDef = getConnectorDef(packageConnectorLib);
+		
+		return connectorDef;
+	}
+	
+	/**
+	 * Retrieves pathmap of a connector library
+	 * 
+	 * @param preferredConnectorType
+	 *          Preferred value of 'ConnectorType' to be used during migration 
+	 *            
+	 * @return pathmap of a connector library which corresponds to the input connectorType value  
+	 *         
+	 */
+	
+	private static String getPathmapConnectorLib(String preferredConnectorType){
+		
+		String pathmapConnectorLib = "";
+		
+		String pathmapAMIConnector = "pathmap://DDS4CCM_AMI_LIBRARIES/CCM_AMI.uml";
+		String pathmapCORBAConnector = "pathmap://DDS4CCM_CORBA_LIBRARIES/CCM_CORBA.uml";
+		
+		if(preferredConnectorType.equals(ConnectorType.AMI4CCM_Connector.name())){
+			pathmapConnectorLib = pathmapAMIConnector; 
+		}else if(preferredConnectorType.equals(ConnectorType.CORBA4CCM_Connector.name())){
+			pathmapConnectorLib = pathmapCORBAConnector;
+		}
+		
+		return pathmapConnectorLib;
+	}
+	
+	/**
+	 * Import a package represented by the input 'packageUri' in the input 'model' and return it. 
+	 * If the package is already imported, return the imported package.  
+	 * 
+	 * @param model
+	 *          model object where the package should be imported
+	 * @param packageUri
+	 * 			URI of the package object that is being imported          
+	 *            
+	 * @return the imported package object  
+	 *         
+	 */
+	private static Package getImportedConnectorPackage(Model model, URI packageUri){
+		
+		Package importedConnectorPackage = null;
+		boolean isAlreadyImported = false;
+	
+		List<PackageImport> listPackageImport = model.getPackageImports();
+		
+		for(PackageImport pi: listPackageImport){
+			
+			if(pi.getImportedPackage().eResource().getURI().equals(packageUri)){
+				isAlreadyImported = true;
+				importedConnectorPackage = pi.getImportedPackage();
+				break;
+			}
+		}
+		
+		if(!isAlreadyImported){
+			org.eclipse.emf.ecore.resource.Resource connectorPackageResource = model.eResource().getResourceSet().createResource(packageUri);			
+			
+			try {
+				connectorPackageResource.load(null);
+				importedConnectorPackage = (Package) connectorPackageResource.getContents().get(0);
+				importPackage(model, importedConnectorPackage);
+			} catch (IOException e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
+			}			
+		}		
+		return importedConnectorPackage;		
+	}
+	
+	private static void importPackage(final Model model, final Package packageObj){
+		
+		org.eclipse.emf.ecore.resource.Resource modelResource = model.eResource();
+		final Package root = (Package) modelResource.getContents().get(0);
+				
+		TransactionalEditingDomain domain =	TransactionUtil.getEditingDomain(model);	
```

Comment:
I don't understand why you are going through all the trouble of creating a RecordingCommand to create the package import. Presumably the entire migration is already wrapped in a RecordingCommand, so the package import will also be.